templates/validationMultipart.hbs

```diff
@@ -0,0 +1,130 @@
+const {{constantName (concat nativeType 'Keys')}}: string[] = [{{#each properties}}{{{stringLiteral serializedName}}}{{#hasMore}}, {{/hasMore}}{{/each}}]
+
+function model{{className nativeType.parentType}}FromMultipartFormDataContent(name: string, value: any, knownKeys: Record<string, boolean> = {}): {{#if discriminator}}Omit<{{nativeType.parentType}}, {{{stringLiteral discriminator.serializedName}}}>{{else}}{{nativeType}}{{/if}} {
+	if (typeof value !== 'object' || value === undefined || value === null) {
+		throw `Invalid type for ${name}: expected object got ${typeof value}`
+	}
+
+	{{constantName (concat nativeType 'Keys')}}.forEach(k => knownKeys[k] = true)
+
+	const result: {{#if discriminator}}Omit<{{nativeType.parentType}}, {{{stringLiteral discriminator.serializedName}}}>{{else}}{{nativeType}}{{/if}} = {
+	{{#each parents}}
+		...{{>frag/fromJson . prefix=''}}Content(name, value, knownKeys),
+	{{/each}}
+	{{#each properties}}
+	{{#unless readOnly}}
+	{{#if required}}
+	{{#if nullable}}
+		{{{stringLiteral serializedName}}}: allowNull({{>frag/fromMultipart schema prefix=''}})(`${name}.{{serializedName}}`, {{>frag/parseMultipartValue .}}),
+	{{else}}
+		{{{stringLiteral serializedName}}}: {{>frag/fromMultipart schema prefix=''}}(`${name}.{{serializedName}}`, {{>frag/parseMultipartValue .}}),
+	{{/if}}
+	{{else}}
+	{{#if nullable}}
+		{{{stringLiteral serializedName}}}: allowNullOrUndefined({{>frag/fromMultipart schema prefix=''}})(`${name}.{{serializedName}}`, {{>frag/parseMultipartValue .}}),
+	{{else}}
+		{{{stringLiteral serializedName}}}: allowUndefined({{>frag/fromMultipart schema prefix=''}})(`${name}.{{serializedName}}`, {{>frag/parseMultipartValue .}}),
+	{{/if}}
+	{{/if}}
+	{{/unless}}
+	{{/each}}
+	{{#each discriminatorValues}}
+		{{{stringLiteral schema.discriminator.serializedName}}}: {{{literalValue}}},
+	{{/each}}
+	}
+
+	return result
+}
+
+function model{{className nativeType.parentType}}ToJsonContent(name: string, value: {{nativeType}}, knownKeys: Record<string, boolean> = {}): ToJson<{{#if discriminator}}Omit<{{nativeType.parentType}}, {{{stringLiteral discriminator.serializedName}}}>{{else}}{{nativeType}}{{/if}}> {
```

Comment:
Do we use these `toJson` functions? Are they different from the ones in the json validation?

Reply:
These functions are the same as those used in JSON validation. I don't believe they get used.